lib/react_client/react_interop.dart

```diff
@@ -352,9 +368,19 @@ class ReactDartInteropStatics2 implements ReactDartInteropStatics {
     dynamic Function(Component2 component) handleGetInitialState,
     void Function(Component2 component, ReactComponent jsThis) handleComponentWillMount,
     void Function(Component2 component) handleComponentDidMount,
-    // TODO: Should this be removed when we update Component2.componentWillReceiveProps to throw an UnsupportedError? (CPLAT-4765)
-    void Function(Component2 component, JsMap jsNextProps) handleComponentWillReceiveProps,
-    bool Function(Component2 component, JsMap jsNextProps, JsMap jsNextState) handleShouldComponentUpdate,
+    void Function(
+      ComponentStatics componentInstance,
```

Comment:
This name isn't right, and this should supply a generic parameter
```suggestion
      ComponentStatics<Component2> componentStatics,
```